staging/src/k8s.io/apiserver/pkg/server/egressselector/egress_selector.go

```diff
@@ -153,34 +156,51 @@ func createConnectTCPDialer(tcpTransport *apiserver.TCPTransport) (utilnet.DialF
 	}
 	contextDialer := func(ctx context.Context, network, addr string) (net.Conn, error) {
 		klog.V(4).Infof("Sending request to %q.", addr)
+		start := time.Now()
```

Comment:
Using `time.Now()` here makes it very hard to test behavior. Alternatively, you can add a clock to egressmetrics.Metrics, so that you can call egressmetrics.Metrics.clock to get the current time (and later to calculate elapsed time). If you make that change, then we can add unit tests to verify that the behavior is correct.